zaza/openstack/charm_tests/hacluster/tests.py

```diff
@@ -69,3 +70,71 @@ def test_920_put_in_maintenance(self):
 
         self._toggle_maintenance_and_wait('true')
         self._toggle_maintenance_and_wait('false')
+
+    def test_930_scaleback_bionic(self):
+        """Remove a unit, recalculate quorum and add a new one."""
+        principle_app = 'keystone'
+        principle_units = zaza.model.get_status().applications[
+            principle_app]['units']
+        self.assertEqual(len(principle_units), 3)
+        doomed_principle = sorted(principle_units.keys())[0]
+        series = juju_utils.get_machine_series(
+            principle_units[doomed_principle].machine)
+        if series != 'bionic':
```

Comment:
My gut feel is that it shouldn't be hard-coded here and instead this should be handled at the tests.yaml level for charm-hacluster.

So what you could do is:

* create a separate test case class deriving from `test_utils.OpenStackBaseTest` which would only contain test_930_scaleback_bionic (such as `class HaclusterScalebackTest(test_utils.OpenStackBaseTest)`);
* Instead of using a default model alias, you could use a separate one and use model-specific configure and test steps (see the links below).
  * the model-specific test step for the bionic bundle would contain HaclusterScalebackTest, while the test steps for the default alias would not.

https://github.com/openstack-charmers/zaza/blob/d341c246ebdf8fe6405f643ee9ccc66f5e5fe6ab/zaza/charm_lifecycle/utils.py#L36-L48
https://github.com/openstack-charmers/zaza/blob/d341c246ebdf8fe6405f643ee9ccc66f5e5fe6ab/zaza/charm_lifecycle/utils.py#L257-L261
https://github.com/openstack-charmers/zaza/blob/d341c246ebdf8fe6405f643ee9ccc66f5e5fe6ab/zaza/charm_lifecycle/utils.py#L208-L215
https://github.com/openstack-charmers/zaza/blob/d341c246ebdf8fe6405f643ee9ccc66f5e5fe6ab/zaza/charm_lifecycle/utils.py#L311-L315